asserts/database.go

```diff
@@ -125,26 +121,86 @@ func (db *Database) GenerateKey(authorityID string) (fingerprint string, err err
 		return "", fmt.Errorf("failed to generate private key: %v", err)
 	}
 
-	return db.ImportKey(authorityID, privKey)
+	return db.ImportKey(authorityID, WrapPrivateKey(privKey))
 }
 
 // ImportKey stores the given private/public key pair for identity and
 // returns its fingerprint
-func (db *Database) ImportKey(authorityID string, privKey *packet.PrivateKey) (fingerprint string, err error) {
-	buf := new(bytes.Buffer)
-	err = privKey.Serialize(buf)
+func (db *Database) ImportKey(authorityID string, privKey PrivateKey) (fingerprint string, err error) {
+	encoded, err := encodePrivateKey(privKey)
 	if err != nil {
 		return "", fmt.Errorf("failed to store private key: %v", err)
 	}
 
-	fingerp := hex.EncodeToString(privKey.PublicKey.Fingerprint[:])
-	err = db.atomicWriteEntry(buf.Bytes(), true, privateKeysRoot, authorityID, fingerp)
+	fingerp := privKey.PublicKey().Fingerprint()
+	err = db.atomicWriteEntry(encoded, true, privateKeysRoot, authorityID, fingerp)
 	if err != nil {
 		return "", fmt.Errorf("failed to store private key: %v", err)
 	}
 	return fingerp, nil
 }
 
+// finding 0 or more than one private key are considered errors
```

Comment:
(nitpick) I had trouble parsing this line, maybe reword a little bit like "findPrivateKey will returns an error if none or more than one private keys are found"?